### ISSUE TITLE
Failing to connect to included PostgreSQL server

### ISSUE THREAD

**Issue by sam-bowman** [netbox-community/netbox-chart#347]
Chart Revision:  5.0.0-beta.94 
Deployed via Helm (through ArgoCD)
Completely base installation of the chart with included Redis and PostgreSQL. Values file contents:
```
debug: true
extraEnvs:
  - name: DB_WAIT_DEBUG
    value: '1'
```

When trying to deploy Netbox into my Kubernetes cluster, I'm getting various errors with the Netbox containers connecting to the PostgreSQL service.

Initially after a completely base deployment, Netbox fails to authenticate against PostgreSQL. It seemed like the PostgreSQL hand't created the 'netbox' user. Deleting the PostgreSQL PVC & Pod and letting them re-create automatically seemed to solve this, after this it no longer gave the auth error & I could see in the PostgreSQL logs that it had created the 'netbox' user.

However, I ran into another issue I've been unable to resolve. The Netbox container only showed the following message in the logs, even with debug on:
`⚙️ Applying database migrations`
It would just show that, then the pod would crash and restart.

The PostgreSQL logs showed the below, with the last 2 lines (the ERROR and STATEMENT lines) repeating over and over as the Netbox container attempts to connect:

```logs
postgresql 14:03:04.36 INFO  ==>
postgresql 14:03:04.43 INFO  ==> Welcome to the Bitnami postgresql container
postgresql 14:03:04.44 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql 14:03:04.47 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
postgresql 14:03:04.56 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
postgresql 14:03:04.57 INFO  ==>
postgresql 14:03:04.98 INFO  ==> ** Starting PostgreSQL setup **
postgresql 14:03:05.37 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 14:03:05.47 INFO  ==> Loading custom pre-init scripts...
postgresql 14:03:05.63 INFO  ==> Initializing PostgreSQL database...
postgresql 14:03:05.87 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql 14:03:05.96 INFO  ==> Generating local authentication configuration
postgresql 14:03:21.46 INFO  ==> Starting PostgreSQL in background...
postgresql 14:03:24.86 INFO  ==> Changing password of postgres
postgresql 14:03:25.27 INFO  ==> Creating user netbox
postgresql 14:03:25.97 INFO  ==> Granting access to "netbox" to the database "netbox"
postgresql 14:03:26.87 INFO  ==> Setting ownership for the 'public' schema database "netbox" to "netbox"
postgresql 14:03:27.57 INFO  ==> Configuring replication parameters
postgresql 14:03:28.16 INFO  ==> Configuring synchronous_replication
postgresql 14:03:28.17 INFO  ==> Configuring fsync
postgresql 14:03:28.97 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped
postgresql 14:03:29.97 INFO  ==> Loading custom scripts...
postgresql 14:03:30.07 INFO  ==> Enabling remote connections
postgresql 14:03:30.26 INFO  ==> ** PostgreSQL setup finished! **
postgresql 14:03:30.57 INFO  ==> ** Starting PostgreSQL **
2024-09-16 14:03:30.784 GMT [1] LOG:  pgaudit extension initialized
2024-09-16 14:03:30.969 GMT [1] LOG:  starting PostgreSQL 16.4 on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024-09-16 14:03:30.970 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024-09-16 14:03:30.970 GMT [1] LOG:  listening on IPv6 address "::", port 5432
2024-09-16 14:03:30.981 GMT [1] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2024-09-16 14:03:30.992 GMT [168] LOG:  database system was shut down at 2024-09-16 14:03:29 GMT
2024-09-16 14:03:31.133 GMT [1] LOG:  database system is ready to accept connections
...
2024-09-16 14:03:35.367 GMT [179] ERROR:  relation "django_content_type" does not exist at character 106
2024-09-16 14:03:35.367 GMT [179] STATEMENT:  SELECT "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" WHERE (("django_content_type"."app_label" = 'auth' AND "django_content_type"."model" IN ('group', 'permission')) OR ("django_content_type"."app_label" = 'contenttypes' AND "django_content_type"."model" IN ('contenttype')) OR ("django_content_type"."app_label" = 'sessions' AND "django_content_type"."model" IN ('session')) OR ("django_content_type"."app_label" = 'social_django' AND "django_content_type"."model" IN ('code', 'association', 'nonce', 'usersocialauth', 'partial')) OR ("django_content_type"."app_label" = 'taggit' AND "django_content_type"."model" IN ('tag', 'taggeditem')) OR ("django_content_type"."app_label" = 'core' AND "django_content_type"."model" IN ('configrevision', 'managedfile', 'datafile', 'autosyncrecord', 'job', 'objectchange', 'datasource')) OR ("django_content_type"."app_label" = 'users' AND "django_content_type"."model" IN ('userconfig', 'token', 'group', 'user', 'objectpermission')) OR ("django_content_type"."app_label" = 'circuits' AND "django_content_type"."model" IN ('circuit', 'provideraccount', 'circuitgroupassignment', 'circuittermination', 'circuitgroup', 'provider', 'providernetwork', 'circuittype')) OR ("django_content_type"."app_label" = 'dcim' AND "django_content_type"."model" IN ('devicebaytemplate', 'inventoryitemtemplate', 'cabletermination', 'consoleport', 'device', 'rack', 'module', 'inventoryitem', 'sitegroup', 'consoleserverport', 'manufacturer', 'powerporttemplate', 'cable', 'devicebay', 'racktype', 'inventoryitemrole', 'consoleporttemplate', 'virtualchassis', 'site', 'rackrole', 'rearport', 'moduletype', 'powerfeed', 'poweroutlettemplate', 'poweroutlet', 'modulebay', 'virtualdevicecontext', 'rackreservation', 'frontporttemplate', 'cablepath', 'powerport', 'devicetype', 'location', 'rearporttemplate', 'platform', 'powerpanel', 'frontport', 'region', 'consoleserverporttemplate', 'modulebaytemplate', 'devicerole', 'interface', 'interfacetemplate')) OR ("django_content_type"."app_label" = 'ipam' AND "django_content_type"."model" IN ('rir', 'routetarget', 'servicetemplate', 'asn', 'iprange', 'vlan', 'role', 'ipaddress', 'prefix', 'vrf', 'fhrpgroupassignment', 'vlangroup', 'service', 'aggregate', 'asnrange', 'fhrpgroup')) OR ("django_content_type"."app_label" = 'extras' AND "django_content_type"."model" IN ('subscription', 'taggeditem', 'dashboard', 'configcontext', 'notificationgroup', 'webhook', 'customlink', 'eventrule', 'stagedchange', 'cachedvalue', 'tag', 'customfield', 'branch', 'bookmark', 'script', 'customfieldchoiceset', 'notification', 'journalentry', 'imageattachment', 'savedfilter', 'configtemplate', 'exporttemplate')) OR ("django_content_type"."app_label" = 'tenancy' AND "django_content_type"."model" IN ('tenantgroup', 'contactgroup', 'tenant', 'contactrole', 'contactassignment', 'contact')) OR ("django_content_type"."app_label" = 'virtualization' AND "django_content_type"."model" IN ('clustertype', 'virtualdisk', 'vminterface', 'clustergroup', 'virtualmachine', 'cluster')) OR ("django_content_type"."app_label" = 'vpn' AND "django_content_type"."model" IN ('ipsecprofile', 'l2vpntermination', 'tunneltermination', 'ikepolicy', 'ipsecproposal', 'tunnelgroup', 'ikeproposal', 'l2vpn', 'ipsecpolicy', 'tunnel')) OR ("django_content_type"."app_label" = 'wireless' AND "django_content_type"."model" IN ('wirelesslangroup', 'wirelesslink', 'wirelesslan')) OR ("django_content_type"."app_label" = 'django_rq' AND "django_content_type"."model" IN ('queue')))
...
```

**Comment by LeoColomb** [comment]
Thanks for opening this issue @sam-bowman.
The error from the PostgreSQL logs are expected while the migrations are not completed.
Enabling `debug` and `dbWaitDebug` introduces a substantial performance penalty.
Which all in all tend to raise the concern regarding the resources available for your deployment to run, very much like https://github.com/netbox-community/netbox-chart/issues/316.

**Comment by sam-bowman** [comment]
Thanks for your help, you reminded me of an issue with my cluster I had been meaning to look at but had completely forgotten, and I think I've found the cause of my issues. 
I had noticed a while back that one of my nodes seemed to be running slowly. Eventually pods etc. would successfully deploy to the node, but it'd take longer than any other node in my cluster. Turns out the Netbox server pods happened to assign themselves to this node (it was likely the lowest utilised one as it also happens to be one of the newest). 
I think I have a storage issue on this node that's slowing down everything. I'm going to do some further testing, and try using node affinity to have the Netbox pods run on another node.
